src/amazonsqs/amazonsqs_connector.bal

```diff
@@ -73,18 +73,17 @@ public type Client client object {
             attributeNumber = attributeNumber + 1;
         }
         http:Request|GeneratePOSTRequestFailed request = generatePOSTRequest(self.accessKey, 
```

Comment:
Can't we move the generatePostRequest function to client object and get rid of the `self.` arguments?

Reply:
Will do it later once the _generatePOSTRequest_ is moved to a common repo location later. This may take a bit time as we have to make sure the other Amazon connectors also can reuse the same piece of code.

Reply:
I think in this iteration we can move the generatePostRequest method to the client connector object. The problem we are solving there is orthogonal to the Amazon commons connector story. There is another method in amazon commons that does the headers and payload signing. You can invoke that method inside the generatePostRequest method once we sort that out. And we need an issue to track that as well (Most probably we will forget about it).

Reply:
Fixed